lib/matplotlib/cm.py

```diff
@@ -371,8 +372,9 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
                 else:
                     raise ValueError("Image RGB array must be uint8 or "
                                      "floating point; found %s" % xx.dtype)
-                # Account for any masked entries in the original array
+                # Account for any nans or masked entries in the original array
                 # If any of R, G, B, or A are masked for an entry, we set alpha to 0
+                xx[np.any(np.isnan(x), axis=2), 3] = 0
```

Comment:
We might want to move this up into only the floating point portion so we aren't checking for nan's on non-float data.